### PR TITLE
Added harmonized occupation group variables to variables and variables details sheet

### PR DESCRIPTION
**Pull request by yulric**
# Changelog

## Features

* Added occupation group variable **LBFA_31A** which is recoded from the variable of the same name in CCHS 2001
* Added occupation group variable **LBFA_31A_a** which is recoded from LBFA_31A from CCHS 2001 and LBSGSOC from CCHS 2007 to 2014 using the same coding as this paper https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6423929/
* Added new occupation group variable **LBFA_31A_b** which is recoded the exact same way as LBFA_31A_a except that the "Other" category in CCHS 2001 is recoded to a new "Other" category rather than missing

### CODE REVIEW

File: inst/extdata/variable_details.csv
Comment (wyusuf068) [inline inst/extdata/variable_details.csv:1557]: I tend to keep CCHS variables in alphabetical order just for ease of access to the user.

**Review by wyusuf068 — APPROVED** [review]

**Comment by DougManuel** [comment]
Don't we also want to harmonize CCHS 2001 and 2007 to 2014 into a common variable, as per https://github.com/Big-Life-Lab/cchsflow/issues/54 and https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6423929/. 

I think this means that we can transform the nine categories from LBFA_31A in 2001 directly to the five variables in LBSGSOC. 

I've made the suggested changes. Please check. The paper by Nowrouzi-Kia et al. included Stata recoding recode info 👍 but there was missing recoding for 2001 category 9 "other" 😬. Suggest recoding from 9 "other" to  "na::b" "missing"?

**Comment by DougManuel** [comment]
I updated my commit for a one-row-per-category approach. @wyusuf068 and I discussed this and I think it makes sense. Maybe not as important for 96, 97, 98, 99 that are pretty standard re-codes.

**Review by DougManuel — CHANGES_REQUESTED** [review]
See the comments below about suggested changes. I made these changes but they require a good review by @yulric  and @wyusuf068

**Review by wyusuf068 — CHANGES_REQUESTED** [review]
1. I'm not sure if respondents who answered "Other" in the 2001 cycle should fall in the missing category. Perhaps add an "Other" category in `LBSGSOC`? This would mean having 6 valid categories as opposed to 5.
2. Given the transformations done to harmonize `LBFA_31A` into `LBSGSOC`, I believe roxygen documentation is necessary to explain to users how the 2001 occupation variable was transformed,  citing the transformations done by Nowrouzi-Kia et al.

File: inst/extdata/variable_details.csv
Comment (yulric) [reply inst/extdata/variable_details.csv]: Yeah, sorry. I was going to do that after it got approved so that the PR would not be hard to review.

**Comment by yulric** [comment]
1. I think "Professional (including accountants)" should be mapped to "Business, Finance, Admin".
2. Not sure what to map "Technologist, Technician or Tech Occ" since there is no mention of tech in the later CCHS. Same with the "Farming, Forestry, Fishing, Mining".

**Comment by DougManuel** [comment]
It sounds like we have concerns about the transformation `LBFA_31A` into `LBSGSOC` crosses into the grey zone of possible concern. 

What about:
1) Creating a new variable for the Nowrouzi-Kai variable. Something like `LBFA_31A_a` that follows their approach. I _think_ what I've created does that but it requires checking. 
2) Creating a second new variable `LBFA_31A_b` that follows what we think is the best practice or practice of cchsflow. This could be a six category with `other`. 

It seems odd that there isn't an 'other' category in `LBSGSOC` since there are professions that would be difficult for respondents to classify themselves within the `LBSGSOC` categories. We may need to investigate further. 

We could make those changes above and then connect with Nowrouzi-Kai group to ask their opinion whether we've correctly captured their work and also for additional insights. I could also connect with the relevant folks at Statscan to see if I can find the discussion for recategorizing in 2007. We may also want to reach out Cam Mustard and colleagues at IWH who have some experience with these variables.

**Comment by DougManuel** [comment]
@wyusuf068 I agree that this is a case for rOxygen documentation. I'm inclined to finish this PR with our current best understanding of the transformation (meaning follow Nowarouzi-Kai approach) and then update later after we investigate further. We could add a note (caution) in the documentation referencing this PR and indicate that this variable may be further updated in near future.

**Comment by yulric** [comment]
I've updated the sheets as per your recommendations Doug.
I've also updated the PR description to add a Changelog, so when we publish a new version of cchsflow we can copy paste that as part of the overall one. I've recently beein thinking its good practice to do this, makes releasing new version down the line easier and makes the PR easier.

**Comment by DougManuel** [comment]
Should we leave your issues for a later revision? If so, let's close this PR and merge to dev. I've been adding the changelog when merging to dev, but this could be done later. Good suggestion to add in the PR.

```
I think "Professional (including accountants)" should be mapped to "Business, Finance, Admin".
Not sure what to map "Technologist, Technician or Tech Occ" since there is no mention of tech in the later CCHS. Same with the "Farming, Forestry, Fishing, Mining".
```

**Comment by yulric** [comment]
> Should we leave your issues for a later revision? If so, let's close this PR and merge to dev. I've been adding the changelog when merging to dev, but this could be done later. Good suggestion to add in the PR.
> 
> ```
> I think "Professional (including accountants)" should be mapped to "Business, Finance, Admin".
> Not sure what to map "Technologist, Technician or Tech Occ" since there is no mention of tech in the later CCHS. Same with the "Farming, Forestry, Fishing, Mining".
> ```

I think we should map "Professional (including accountants)" to "Business, Finance, Admin".
"Technologist, Technician or Tech Occ" should be its own category since its not present in any other CCHS cycles.
"Farming, Forestry, Fishing, Mining" can be put into the category with "Primary Industry".

If you agree with that I'll make those changes to the LBFA_31A_b variable.

**Comment by yulric** [comment]
@DougManuel @wyusuf068 @CBjerke Can I merge this in?

**Comment by DougManuel** [comment]
Yes.

**Pull request merged**
